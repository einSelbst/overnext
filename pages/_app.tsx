// import App from "next/app";
import { AppProps /*, AppContext */ } from 'next/app'

// Conditionally inject axe into the page.
// This only happens outside of production and in a browser (not SSR).
// @see: https://github.com/dequelabs/axe-core-npm/tree/develop/packages/react/examples/next.js
// TODO: why not use next.js dynamic import?
if (typeof window !== 'undefined' && process.env.NODE_ENV !== 'production') {
  const React = require('react')
  const ReactDOM = require('react-dom')
  const axe = require('@axe-core/react')
  axe(React, ReactDOM, 1000)
}

function MyApp ({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />
}

// Only uncomment this method if you have blocking data requirements for
// every single page in your application. This disables the ability to
// perform automatic static optimization, causing every page in your app to
// be server-side rendered.
//
// MyApp.getInitialProps = async (appContext: AppContext) => {
//   // calls page's `getInitialProps` and fills `appProps.pageProps`
//   const appProps = await App.getInitialProps(appContext);

//   return { ...appProps }
// }

export default MyApp
