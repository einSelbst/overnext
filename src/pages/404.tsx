const Custom404 = (): JSX.Element => {
  return (
    <>
      <h1>404 - Page Not Found</h1>
      <p>
        OMG what happened? Yesterday everything was working fine and now
        it&apos;s broken!?
      </p>
    </>
  )
}

export default Custom404
