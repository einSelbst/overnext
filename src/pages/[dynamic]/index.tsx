import {
  GetStaticPaths,
  GetStaticProps,
  /* GetStaticPathsContext, */
  /* GetStaticPropsContext, */
  InferGetStaticPropsType,
} from 'next'
import React from 'react'

export const config = {
  amp: 'hybrid',
}

const DynamicPage = (
  props: InferGetStaticPropsType<typeof getStaticProps>
): JSX.Element => {
  return (
    <div>
      <h1>DynamicPage Component {props.dynamic}</h1>
      <h2>Locale: {props.locale}</h2>
    </div>
  )
}

// This function gets called at build time on server-side.
// It may be called again, on a serverless function, if
// revalidation is enabled and a new request comes in
export const getStaticProps: GetStaticProps = async ({ locale }) => {
  // Call an external API endpoint to get posts.
  // You can use any data fetching library
  //const res = await fetch(`https://.../posts?locale=${locale}`)
  //const posts = await res.json()

  const res = await Promise.resolve('Hello')

  return {
    props: {
      dynamic: res,
      locale: locale,
    },
    // Next.js will attempt to re-generate the page:
    // - When a request comes in
    // - At most once every second
    //revalidate: 1, // In seconds
  }
}

/**
 * see: https://github.com/vercel/commerce/blob/master/pages/%5B...pages%5D.tsx
 * combine paths with locales so that all pages can be pre-rendered
 */
export const getStaticPaths: GetStaticPaths = async ({ locales }) => {
  await Promise.resolve('async needs await')
  // the paths would come from an api, but to have something to work with:
  let paths = Array.from({ length: 10 }, (_, index) => ({
    params: {
      dynamic: `page-${index}`,
    },
  }))

  if (locales) {
    const pathsI18n = paths.map(path => {
      return locales.map(locale => {
        return { params: path.params, locale: locale }
      })
    })
    paths = pathsI18n.reduce((a, v) => a.concat(v), paths)
  }

  return {
    paths: paths,
    /* paths: [
     *   { params: { dynamic: 'page-1' } },
     *   { params: { dynamic: 'page-1' }, locale: 'en' },
     *   { params: { dynamic: 'page-1' }, locale: 'fr' },
     * ], */
    // Fallback shouldn't be enabled here or otherwise this route
    // will catch every page, even 404s, and we don't want that
    fallback: false,
  }
}

export default DynamicPage
