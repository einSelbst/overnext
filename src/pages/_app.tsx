import type { NextLayoutComponentType, NextPageContext } from 'next'
import { DefaultSeo } from 'next-seo'
import { ThemeProvider } from 'next-themes'
import { useAmp } from 'next/amp'
import dynamic from 'next/dynamic'
import type { ErrorProps } from 'next/error'
import Head from 'next/head'

import ENV from 'config/env.config'
import SEO from 'config/seo.config'
/* import DefaultLayout from 'layouts/default.layout' */
import HolyGrailLayout from 'layouts/holy-grail.layout'
import SiteLayout from 'layouts/site.layout'
import 'styles/global.scss' // eslint-disable-line import/no-unassigned-import
/* import 'styles/main.css' // tailwind */

const sendMetric = async ({
  name,
  value,
}: {
  readonly name: string
  readonly value: string
}): Promise<boolean> => {
  if (process.env.NEXT_PUBLIC_QUICK_METRICS_API_KEY === undefined) {
    return false
  }

  // eslint-disable-next-line no-console
  console.log('sending metrics to quickmetrics...')

  // values must be integers
  const clsMultiplier = 1000
  const valueInt: number = Math.round(
    name === 'CLS'
      ? Number.parseFloat(value) * clsMultiplier
      : Number.parseFloat(value)
  )
  const url = `https://qckm.io?m=${name}&v=${valueInt}&k=${process.env.NEXT_PUBLIC_QUICK_METRICS_API_KEY}`

  // Use `navigator.sendBeacon()` if available, falling back to `fetch()`.
  if (typeof navigator.sendBeacon === 'function') {
    navigator.sendBeacon(url)
  } else {
    await fetch(url, { keepalive: true, method: 'POST' })
  }
  return true
}

/**
 * Web Vitals
 * @see {@link https://www.freecodecamp.org/news/how-to-measure-next-js-web-vitals-using-quickmetrics/}
 */
const reportWebVitals = async (metric: {
  readonly name: string
  readonly value: string
}): Promise<void> => {
  // I can only send 5 metrics to free quickmetrics account
  if (metric.name !== 'Next.js-hydration') {
    await sendMetric(metric)
  }
  console.log(metric) // eslint-disable-line no-console
}

/**
 * Conditionally inject axe into the page.
 * This only happens outside of production and in a browser (not SSR).
 * @see {@link https://github.com/dequelabs/axe-core-npm/tree/develop/packages/react/examples/next.js}
 *
 * Regarding next.js dynamic
 * @see {@link https://github.com/vercel/next.js/blob/0af3b526408bae26d6b3f8cab75c4229998bf7cb/examples/with-dynamic-import/pages/index.js }
 */
const A11yLinter = dynamic(
  /* eslint-disable import/max-dependencies */
  import(
    /* webpackChunkName: "axeHelper" */
    'components/helper/axe'
  ),
  /* eslint-enable import/max-dependencies */
  { ssr: false }
)

const HeadIcons = (): JSX.Element => {
  const isNotAmp = !useAmp()

  return (
    <Head>
      {isNotAmp && (
        <>
          <meta
            content='initial-scale=1.0, width=device-width'
            name='viewport'
          />
          <link href='https://fonts.gstatic.com' rel='preconnect' />
          {/* the preload is automatically added by next.js or one of my linter rules */}
          <link
            as='style'
            href='https://fonts.googleapis.com/css2?family=Inter&family=Sansita&display=swap'
            rel='preload'
          />
          <link
            href='https://fonts.googleapis.com/css2?family=Inter&family=Sansita&display=swap'
            rel='stylesheet'
          />
          {/* Safari & Chrome 93 supports media in 'theme-color' so I can use different */}
          {/* theme colors based on color scheme. However, this doesn't work with AMP. */}
          <meta
            content='white'
            media='(prefers-color-scheme: light)'
            name='theme-color'
          />
          <meta
            content='darkgrey'
            media='(prefers-color-scheme: dark)'
            name='theme-color'
          />
        </>
      )}
      {useAmp() && <meta content='#ffffff' name='theme-color' />}

      {/* <meta name="viewport" content="width=device-width, initial-scale=0.86, maximum-scale=5.0, minimum-scale=0.86" /> see https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag */}
      {/* Browsers use this in some areas to help your brand feel more embedded */}

      {/* <link
          rel='preload'
          href='https://unpkg.com/marx-css/css/marx.min.css'
          as='style'
          type='text/css'
          />
          <link
          rel='stylesheet'
          href='https://unpkg.com/marx-css/css/marx.min.css'
          type='text/css'
          />
          <link rel="stylesheet" href="node_modules/modern-normalize/modern-normalize.css" />
        */}

      {/* Windows uses these to help your brand feel more embedded */}
      <meta content='#da532c' name='msapplication-TileColor' />
      <meta
        content='/icons/mstile-150x150.png'
        name='msapplication-TileImage'
      />
      {/* Browsers use these as tab and app icons */}
      <link href='/favicon.ico' rel='icon' />
      <link href='/favicon.ico' rel='shortcut icon' type='image/x-icon' />
      <link href='/icons/apple-touch-icon.png' rel='apple-touch-icon' />
      <link
        href='/icons/icon-192x192.png'
        rel='icon'
        sizes='192x192'
        type='image/png'
      />
      <link
        href='/icons/icon-32x32.png'
        rel='icon'
        sizes='32x32'
        type='image/png'
      />
      <link
        href='/icons/icon-16x16.png'
        rel='icon'
        sizes='16x16'
        type='image/png'
      />
      <link
        href='/icons/apple-touch-icon-57x57.png'
        rel='apple-touch-icon'
        sizes='57x57'
      />
      <link
        href='/icons/apple-touch-icon-72x72.png'
        rel='apple-touch-icon'
        sizes='72x72'
      />
      <link
        href='/icons/apple-touch-icon-76x76.png'
        rel='apple-touch-icon'
        sizes='76x76'
      />
      <link
        href='/icons/apple-touch-icon-114x114.png'
        rel='apple-touch-icon'
        sizes='114x114'
      />
      <link
        href='/icons/apple-touch-icon-120x120.png'
        rel='apple-touch-icon'
        sizes='120x120'
      />
      <link
        href='/icons/apple-touch-icon-144x144.png'
        rel='apple-touch-icon'
        sizes='144x144'
      />
      <link
        href='/icons/apple-touch-icon-152x152.png'
        rel='apple-touch-icon'
        sizes='152x152'
      />
      <link
        href='/icons/apple-touch-icon.png'
        rel='apple-touch-icon'
        sizes='180x180'
      />
      <link
        color='#5bbad5'
        href='/icons/safari-pinned-tab.svg'
        rel='mask-icon'
      />
      {/* for PWA: */}
      {/* <link rel="manifest" href="path/to/manifest.json" crossorigin="use-credentials" /> */}
      <link href='/manifest.json' rel='manifest' />
    </Head>
  )
}

/**
 * Now here I do mix 2 different ways for adding a Layout, the simpler version
 * where a page has a property `Layout` and the slightly more advanced version
 * where a page has a function `getLayout`. I will keep both for the moment.
 */
const _app = ({
  Component,
  pageProps,
  err,
}: {
  readonly Component: NextLayoutComponentType
  readonly pageProps: NextPageContext
  readonly err: ErrorProps
}): React.ReactNode => {
  /* }): JSX.Element => { */
  const isAmp = useAmp()

  /*
   * Layout specific
   * every page can specify its layout, if none is specifid we define a fallback here
   */
  const Layout = Component.Layout ?? HolyGrailLayout
  const withSiteLayout = (page: React.ReactNode): React.ReactNode => (
    <SiteLayout>{page}</SiteLayout>
  )
  const getLayout = Component.getLayout ?? withSiteLayout

  /* eslint-disable react/jsx-props-no-spreading */
  /* amp pages doesn't work well with the theme so I had to split it */
  return isAmp ? (
    <>
      <DefaultSeo {...SEO} />
      <HeadIcons />
      <Component {...pageProps} err={err} />
    </>
  ) : (
    <ThemeProvider attribute='class' defaultTheme='system'>
      <DefaultSeo {...SEO} />
      <HeadIcons />
      {/* <Layout {...this.props}> */}
      <Layout>{getLayout(<Component {...pageProps} err={err} />)}</Layout>

      {/* <Layout>
            <Component {...pageProps} />
            </Layout>
          */}
      {/* <LayoutTree
            Component={Component}
            pageProps={pageProps} /> */}

      {process.browser && ENV.DEVELOPMENT && <A11yLinter />}
    </ThemeProvider>
  )
  /* eslint-enable react/jsx-props-no-spreading */
}

export { reportWebVitals }
export default _app
