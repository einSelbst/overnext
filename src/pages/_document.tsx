import Document, {
  DocumentContext,
  Html,
  Head,
  Main,
  NextScript,
  DocumentInitialProps,
} from 'next/document'

class _document extends Document {
  static async getInitialProps (
    ctx: DocumentContext
  ): Promise<DocumentInitialProps> {
    return await Document.getInitialProps(ctx)
  }

  render (): JSX.Element {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default _document
