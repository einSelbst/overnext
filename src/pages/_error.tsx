import { captureException, flush } from '@sentry/nextjs'
import type { NextPageContext } from 'next'
import type { ErrorProps } from 'next/error'
import NextErrorComponent from 'next/error'

// extending the nextjs base error type
type ErrorComponentProps = ErrorProps & {
  err?: Error
  hasGetInitialPropsRun?: boolean
}

/**
 * <ErrorPage />
 *
 * @see https://github.com/natterstefan/next-with-sentry/blob/main/pages/_error.js
 */
const _error = ({
  err,
  hasGetInitialPropsRun,
  statusCode,
  title,
}: Readonly<ErrorComponentProps>): ComponentReturnType => {
  if (hasGetInitialPropsRun === false && Boolean(err)) {
    /*
     * getInitialProps is not called in case of
     * https://github.com/vercel/next.js/issues/8592. As a workaround, we pass
     * err via _app.js so it can be captured
     */
    captureException(err)
    // Flushing is not required in this case as it only happens on the client
  }
  return <NextErrorComponent statusCode={statusCode} title={title} />
}

/**
 * Add logger
 */
_error.getInitialProps = async (
  context: NextPageContext
): Promise<ErrorComponentProps> => {
  const errorInitialProps = (await NextErrorComponent.getInitialProps(
    context
  )) as ErrorComponentProps
  const { err, asPath } = context

  /*
   * Workaround for https://github.com/vercel/next.js/issues/8592, mark when
   * getInitialProps has run
   */
  errorInitialProps.hasGetInitialPropsRun = true

  /*
   * Running on the server, the response object (`res`) is available.
   *
   * Next.js will pass an err on the server if a page's data fetching methods
   * threw or returned a Promise that rejected
   *
   * Running on the client (browser), Next.js will provide an err if:
   *
   *  - a page's `getInitialProps` threw or returned a Promise that rejected
   *  - an exception was thrown somewhere in the React lifecycle (render,
   *    componentDidMount, etc) that was caught by Next.js's React Error
   *    Boundary. Read more about what types of exceptions are caught by Error
   *    Boundaries: https://reactjs.org/docs/error-boundaries.html
   */

  if (err) {
    captureException(err)
    /* Sentry.addBreadcrumb({ */
    /* data: { */
    /* asPath, */
    /* }, */
    /* }) */

    /*
     * Flushing before returning is necessary if deploying to Vercel, see
     * https://vercel.com/docs/platform/limits#streaming-responses
     */
    await flush(2000)

    return errorInitialProps
  }

  /*
   * If this point is reached, getInitialProps was called without any
   * information about what the error might be. This is unexpected and may
   * indicate a bug introduced in Next.js, so record it in Sentry
   */
  captureException(
    new Error(
      `_error.js getInitialProps missing data at path: ${
        asPath ?? 'unknown error path'
      }`
    )
  )
  await flush(2000)

  return errorInitialProps
}

export default _error
