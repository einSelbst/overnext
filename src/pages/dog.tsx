/* import { GetStaticProps, GetStaticPropsContext } from 'next' */
import { captureMessage } from '@sentry/nextjs'
import type {
  GetStaticProps,
  GetStaticPropsContext,
  GetStaticPropsResult,
  InferGetStaticPropsType,
} from 'next'
import { useAmp } from 'next/amp'
import Head from 'next/head'
import Link from 'next/link'
import { useRouter } from 'next/router'

import Byline from 'components/byline'
import DefaultLayout from 'layouts/default.layout'
import type { NextLayoutPage } from 'typings/layout'

/* eslint-disable import/exports-last, import/group-exports */
/* istanbul ignore next */
export const config = {
  amp: 'hybrid',
}
/* eslint-enable import/exports-last, import/group-exports */

type HomeProps = {
  host: string
}

const getStaticProps: GetStaticProps = async (
  _context: GetStaticPropsContext
): Promise<GetStaticPropsResult<HomeProps>> => {
  await Promise.resolve(
    'This is just a placeholder to make my typescript linter happy'
  )
  /* eslint-disable no-console */
  /*
   * console.log('now in getstaticprops')
   *
   * console.log('process.env.URL')
   * console.log(process.env.URL)
   *
   * console.log('process.env.SITE_URL')
   * console.log(process.env.SITE_URL)
   */
  let url = 'https://overnext.vercel.app'
  if (typeof process.env.URL !== 'undefined') {
    url = process.env.URL
  }
  console.log(url)
  /* eslint-enable no-console */

  return {
    props: {
      host: url,
    },
  }
}

const throwError = () => {
  /* throw new Error('Sentry Frontend Error') */
  /* Sentry.captureException(err); */
  captureMessage('Something went wrong with the dog')
}

/* const Dog:NextPage<HomeProps> = (props: HomeProps): JSX.Element => { */
/* const Dog: NextPage = (props: HomeProps): JSX.Element => { */
/* const Dog: NextPage = (props: HomeProps) => { */
/* const Dog: NextPage = () => { */
const Dog: NextLayoutPage = (
  /* props: InferGetStaticPropsType<typeof getStaticProps> */
  { host }: Readonly<InferGetStaticPropsType<typeof getStaticProps>>
) => {
  /* const Dog = (props: HomeProps): JSX.Element => { */
  const isAmp = useAmp()
  const router = useRouter()
  const { locale } = router
  const imageUrl = 'https://picsum.photos/id/237/800/450'

  return (
    <main>
      <Head>
        <title>The Dog</title>
        {/* preload only the LCP image */}
        <link as='image' href={imageUrl} rel='preload' />
      </Head>
      <h1>The Dog (Hybrid AMP Page)</h1>
      <h2>
        Find out more <a href={host as string}>here</a>
      </h2>
      <Byline author='Meow Meow Fuzzyface' />
      The value of platform is: {process.env.platform}
      The value of customKey is: {process.env.customKey}
      {isAmp ? (
        <amp-img alt='a cute pups' height='450' src={imageUrl} width='800' />
      ) : (
        <img alt='a cute pups' height='450' src={imageUrl} width='800' />
      )}
      <p>
        {isAmp ? (
          <Link href='/dog'>
            <a>View Non-AMP Version</a>
          </Link>
        ) : (
          <a
            href={
              locale === undefined ? '/dog/?amp=1' : `/${locale}/dog?amp=1`
            }>
            view {locale} amp version
          </a>
        )}
      </p>
      <button type='button' onClick={throwError}>
        Throw error
      </button>
      <p className='caption'>Woooooooooooof</p>
      <p>
        The redirect from the non-amp version to the amp version is not working
        when using LINK but without using LINK I don&apos;t get it localized. I
        think the reason is that link is doing a local/frontend redirect which
        is not able to load the amp stuff.
      </p>
      <p>
        Wafer donut candy soufflé{' '}
        <a href={isAmp ? '/?amp=1' : '/'}>lemon drops</a> icing. Marzipan gummi
        bears pie danish lollipop pudding powder gummi bears sweet. Pie sweet
        roll sweet roll topping chocolate bar dragée pudding chocolate cake.
        Croissant sweet chocolate bar cheesecake candy canes. Tootsie roll icing
        macaroon bonbon cupcake apple pie candy canes biscuit candy canes.
        Jujubes jelly liquorice toffee gingerbread. Candy tootsie roll macaroon
        chocolate bar icing sugar plum pie. Icing gummies chocolate bar
        chocolate marzipan bonbon cookie chocolate tart. Caramels danish halvah
        croissant. Cheesecake cookie tootsie roll ice cream. Powder dessert
        carrot cake muffin tiramisu lemon drops liquorice topping brownie.
        Soufflé chocolate cake croissant cupcake jelly.
      </p>
      <p>
        Muffin gummies dessert cheesecake candy canes. Candy canes danish cotton
        candy tart dessert powder bear claw marshmallow. Muffin chocolate
        marshmallow danish. Chocolate bar biscuit cake tiramisu. Topping sweet
        brownie jujubes powder marzipan. Croissant wafer bonbon chupa chups cake
        cake marzipan caramels jujubes. Cupcake cheesecake sweet roll
        marshmallow lollipop danish jujubes jelly icing. Apple pie chupa chups
        lollipop jelly-o cheesecake jelly beans cake dessert. Tootsie roll
        tootsie roll bonbon pastry croissant gummi bears cake cake. Fruitcake
        sugar plum halvah gingerbread cookie pastry chupa chups wafer lemon
        drops. Marshmallow liquorice oat cake lollipop. Lemon drops oat cake
        halvah liquorice danish powder cupcake soufflé. Cake tart topping
        jelly-o tart sugar plum. Chocolate bar cookie wafer tootsie roll candy
        cotton candy toffee pie donut.
      </p>
      <p>
        Ice cream lollipop marshmallow tiramisu jujubes croissant. Bear claw
        lemon drops marzipan candy bonbon cupcake powder. Candy canes cheesecake
        bear claw pastry cake donut jujubes. Icing tart jelly-o soufflé bonbon
        apple pie. Cheesecake pie chupa chups toffee powder. Bonbon lemon drops
        carrot cake pudding candy halvah cheesecake lollipop cupcake. Pudding
        marshmallow fruitcake. Gummi bears bonbon chupa chups lemon drops. Wafer
        dessert gummies gummi bears biscuit donut tiramisu gummi bears brownie.
        Tootsie roll liquorice bonbon cookie. Sesame snaps chocolate bar cake
        croissant chupa chups cheesecake gingerbread tiramisu jelly. Cheesecake
        ice cream muffin lollipop gummies. Sesame snaps jelly beans sweet bear
        claw tart.
      </p>
      <p>
        Sweet topping chupa chups chocolate cake jelly-o liquorice danish.
        Pastry jelly beans apple pie dessert pastry lemon drops marzipan
        gummies. Jelly beans macaroon bear claw cotton candy. Toffee sweet
        lollipop toffee oat cake. Jelly-o oat cake fruitcake chocolate bar
        sweet. Lemon drops gummies chocolate cake lollipop bear claw croissant
        danish icing. Chocolate bar donut brownie chocolate cake lemon drops
        chocolate bar. Cake fruitcake pudding chocolate apple pie. Brownie
        tiramisu chocolate macaroon lemon drops wafer soufflé jujubes icing.
        Cheesecake tiramisu cake macaroon tart lollipop donut. Gummi bears
        dragée pudding bear claw. Muffin cake cupcake candy canes. Soufflé candy
        canes biscuit. Macaroon gummies danish.
      </p>
      <p>
        Cupcake cupcake tart. Cotton candy danish candy canes oat cake ice cream
        candy canes powder wafer. Chocolate sesame snaps oat cake dragée
        cheesecake. Sesame snaps marshmallow topping liquorice cookie
        marshmallow. Liquorice pudding chocolate bar. Cake powder brownie
        fruitcake. Carrot cake dessert marzipan sugar plum cupcake cheesecake
        pastry. Apple pie macaroon ice cream fruitcake apple pie cookie. Tootsie
        roll ice cream oat cake cheesecake donut cheesecake bear claw. Sesame
        snaps marzipan jelly beans chocolate tootsie roll. Chocolate bar donut
        dragée ice cream biscuit. Pie candy canes muffin candy canes ice cream
        tiramisu.
      </p>
    </main>
  )
}

Dog.Layout = DefaultLayout

export { getStaticProps } // eslint-disable-line import/group-exports
export default Dog
