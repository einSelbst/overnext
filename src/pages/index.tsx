import Head from 'next/head'
import Link from 'next/link'

export const Home = (): JSX.Element => {
  return (
    <div className='container'>
      <Head>
        <title>The Egg</title>
        <meta name='description' content='A page about life on earth' />
      </Head>

      <main>
        <h1>
          Welcome to <a href='https://nextjs.org'>Overnext!</a>
        </h1>

        <p className='description'>
          An overly opinonated starter for <code>next.js</code>
        </p>

        <div className='py-20'>
          <h2 className='text-3xl text-center text-gray-700 dark:text-gray-100'>
            A short story
          </h2>
          <iframe
            width='560'
            height='315'
            src='https://www.youtube-nocookie.com/embed/h6fcK_fRYaI'
            title='YouTube video player'
            allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
            allowFullScreen
          ></iframe>
        </div>

        <button
          data-cy='test-button'
          type='button'
          onClick={() => {
            window.alert('With typescript and Jest')
          }}
        >
          Test Button
        </button>

        <div className='grid'>
          <a href='https://nextjs.org/docs' className='card'>
            <h3>Documentation &rarr;</h3>
            <p>Find in-depth information about Next.js features and API.</p>
          </a>

          <a href='https://nextjs.org/docs/getting-started' className='card'>
            <h3>Learn &rarr;</h3>
            <p>Learn about Next.js in an interactive course with quizzes!</p>
          </a>

          <Link href='/legal' locale='fr'>
            <a href='fr/legal' data-cy='fr-legal' className='card'>
              <h3>Legal &rarr;</h3>
              <p>Discover and enjoy the depths of legal issues.</p>
            </a>
          </Link>

          <a href='https://vercel.com/import?filter=next.js' className='card'>
            <h3>Deploy &rarr;</h3>
            <p>
              Instantly deploy your Next.js site to a public URL with Vercel.
            </p>
          </a>
        </div>
      </main>
      <style jsx>{`
        button {
          background: #0072ed;
        }
      `}</style>
      <style jsx global>{`
        html,
        body {
          color: #000;
          background: #fff;
        }

        [data-theme='dark'],
        [data-theme='dark'] body {
          color: #fff;
          background: #000;
        }

        iframe {
          border: 0;
        }
      `}</style>
    </div>
  )
}

export default Home
