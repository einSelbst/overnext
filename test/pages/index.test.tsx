import * as React from 'react'
import { Home } from '../../src/pages/index'
import { render, fireEvent } from '../test-utils'

describe('home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    // inline-snapshots require 'prettier', but I use 'prettierx'
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Home />, {})
    jest.spyOn(window, 'alert').mockImplementation()
    fireEvent.click(getByText('Test Button'))
    expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  })
})
