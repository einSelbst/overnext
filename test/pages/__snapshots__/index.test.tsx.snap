// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      id="main"
    >
      <h1>
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Overnext!
        </a>
      </h1>
      <p
        class="description"
      >
        An overly opinionated starter for 
        <code>
          next.js
        </code>
      </p>
      <div
        class="py-20"
      >
        <h2
          class="text-3xl text-center text-gray-700 dark:text-gray-100"
        >
          A short story
        </h2>
        <iframe
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
          allowfullscreen=""
          height="315"
          loading="lazy"
          src="https://www.youtube-nocookie.com/embed/h6fcK_fRYaI"
          title="YouTube video player"
          width="560"
        />
      </div>
      <button
        data-cy="test-button"
        type="button"
      >
        Test Button
      </button>
      <h2>
        Some further info
      </h2>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h3>
            Documentation →
          </h3>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/docs/getting-started"
        >
          <h3>
            Learn →
          </h3>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          data-cy="fr-legal"
          href="/legal"
        >
          <h3>
            Legal →
          </h3>
          <p>
            Discover and enjoy the depths of legal issues.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/import?filter=next.js"
        >
          <h3>
            Deploy →
          </h3>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </div>
    <style>
      
      button {
        background: #0072ed;
      }
    
    </style>
    <style>
      
      html,
      body {
        color: #000;
        background: #fff;
      }

      [data-theme='dark'],
      [data-theme='dark'] body {
        color: #fff;
        background: #000;
      }

      a {
        color: #0072ed;
      }

      iframe {
        border: 0;
      }
    
    </style>
  </div>
</DocumentFragment>
`;
